### PR TITLE
WIP generate test vectors + tests

### PR DESCRIPTION
**Pull request by mixmix**
targets:
- [x] derive-secret
- [x] encrypt
- [x] decrypt

### CODE REVIEW

**Comment by dominictarr** [comment]
@mixmix may I suggest, that instead of hardcoding the vector into a javascript file, instead have one script that generates a JSON file, then another test that checks the library functions do the correct thing? then another implementation can just import the JSON file, and use that in their tests. If it's hard coded, they'll have to hand copy it out of the file, which will be painful to maintain.

for example, ssb-validate uses a pattern like this. there is a [test/generate.js](https://github.com/ssbc/ssb-validate/blob/master/test/generate.js)
and then a [test/data.js](https://github.com/ssbc/ssb-validate/blob/master/test/data.js) that loads the data, and checks that everything is correctly valid or invalid.

**Comment by mixmix** [comment]
Yep that's where I'm headed. There's a test vector over in bix2-spec which is raw json.
Also agree on generate.js file idea

File: package.json [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
   "homepage": "https://github.com/ssbc/private-box2",
   "repository": {
     "type": "git",
     "url": "git://github.com/ssbc/private-box2.git"
   },
+  "scripts": {
+    "generate": "node test/generate.js",
+    "test": "npm run test:js && npm run lint",
+    "test:js": "tape 'test/**/*.test.js' | tap-spec",
+    "lint": "eslint ."
```
Comment (mixmix) [inline package.json]: @dominictarr I've added the most minimal `.eslintconfig` I can.
Basically I ran it and disabled (stupid) rules till everything is passing (e.g. no-camel-case, stupid ++ rules). The purpose of this is to catch unused-variables and dangerous things, not to fix indenting.

My policy here is to disable rules that are in the way (notice we don't bother linting the `test/` folder - `.eslintignore`).

This is an experiment in a compromise which I hope keeps the code easy to maintain (no unused variables accidentally hanging around), while also respecting the need to do layout things to improve readability sometimes (e.g. do indenting and camel_case sometimes, and multi-line spaces)

**Comment by keks** [comment]
While I am not strictly opposed to generate that data, I would like the JSON files to be the source of truth. I don't want to need a working node development environment to generate the files.

**Comment by keks** [comment]
Also, it's much easier to verify the correctness of the JSON files instead of the generator code without JS knowledge. There is a reason RFCs do it that way. From a software engineering perspective, these are table-driven tests and the tables shouldn't be generated during the test.

**Comment by mixmix** [comment]
Agree @keks - you don't need Node, I'm just using Node to make things which I'm putting in the `box2-spec` repo. I'm then referencing the `box2-spec` JSON in my tests.

Ideally when these repos are stable that `box2-spec` repo does not change unless we've all reviewed a change and release a new version.

**Comment by mixmix** [comment]
The main input I need from you is whether the format of those JSON works well for you in GoLang

**Pull request merged**